src/react/ui-elements/EmptyBucket/utils.ts

```diff
@@ -0,0 +1,50 @@
+import { S3 } from 'aws-sdk';
+import { ErrorsList } from './types';
+
+const MAX_KEYS = 1000;
+
+export const paginateData = (
+  data: S3.ObjectIdentifierList,
+  maxKeys = MAX_KEYS,
+) => {
+  const index = Math.ceil(data.length / maxKeys);
+  const page = (index - 1) * maxKeys;
+  const arr = data.slice(page);
+
+  return arr;
+};
+
+export const createDeleteObjectsData = (
+  deleteMarkers?: S3.DeleteMarkers,
+  versions?: S3.ObjectVersionList,
+): S3.ObjectIdentifierList => [
+  ...(deleteMarkers || []).map((d) => ({
+    Key: d.Key || '',
+    VersionId: d.VersionId,
+  })),
+  ...(versions || []).map((v) => ({
+    Key: v.Key || '',
+    VersionId: v.VersionId,
+  })),
+];
+
+export const getUniqueErrorMessages = (
+  errorMessages: string[],
+): ErrorsList[] => {
+  const uniqueMessages: { [key: string]: number } = {};
+
+  errorMessages.forEach((message) => {
+    if (uniqueMessages[message]) {
+      uniqueMessages[message]++;
+    } else {
+      uniqueMessages[message] = 1;
+    }
+  });
+
+  const uniqueArray = Object.keys(uniqueMessages).map((message) => ({
+    message,
+    errorNumbers: uniqueMessages[message],
+  }));
+
```

Comment:
I think this implementation might be simplified thanks to the use of a Set. 

`[...new Set(errorMessages)]` probably does the same